python/lsst/summit/utils/butlerUtils.py

```diff
@@ -521,14 +523,14 @@ def getExpRecordFromDataId(butler: dafButler.Butler, dataId: dict) -> dafButler.
             "exposure", where=where, bind={"dayObs": dayObs, "seq_num": seqNum}, datasets="raw"
         )
 
-    expRecords = set(expRecords)
-    if not expRecords:
+    filteredExpRecords = set(expRecords)
+    if not filteredExpRecords:
         raise LookupError(f"No exposure records found for {dataId}")
-    assert len(expRecords) == 1, f"Found {len(expRecords)} exposure records for {dataId}"
-    return expRecords.pop()
+    assert len(filteredExpRecords) == 1, f"Found {len(filteredExpRecords)} exposure records for {dataId}"
+    return filteredExpRecords.pop()
```

Comment:
Was all that necessary just to make mypy happy because it would change the type to being a `set`? If so, that's fine, but it really does make me unhappy that that sort of thing is necessary. I wonder if there's a better way to make it happy...

Reply:
I will try to redeclare the type....

Reply:
But not working...

Reply:
Now I check we are able to control this error using the allow_redefinition = True (I had it to false)